packages/assets-controllers/src/MultichainAssetsRatesController/MultichainAssetsRatesController.ts

```diff
@@ -508,7 +621,21 @@ export class MultichainAssetsRatesController extends StaticIntervalPollingContro
       for (const { accountId, assets } of accounts) {
         const account = this.#getAccount(accountId);
 
-        const rates = await this.#getUpdatedRatesFor(account, assets);
+        // Skip if the account has no Snap ID since we don't know which Snap to
+        // use.
+        //
+        // FIXME: Instead of using the Snap ID from the account, we should
+        // select the Snap based on the supported scopes defined in the Snaps'
+        // manifest.
+        const snapId = account.metadata.snap?.id;
+        if (!snapId) {
+          continue;
+        }
+
+        const rates = await this.#getUpdatedRatesFor(
+          new Map(assets.map((asset) => [asset, snapId as SnapId])),
```

Comment:
This will create a request per account too. Can we deduplicate requests here too?

Reply:
Good catch, indeed